CI: Test all features

Our CI should run with `cargo test --all-features`, especially in light of #991 (which is now blocking).

I was working on turning Landlock into a feature (so as to be able to enable/disable it in builds and compare performance) at the time, which was the motivation for me trying to do so.

```
Run cargo test --lib --all-features
    Updating crates.io index
   Compiling byteorder v1.5.0
   Compiling llvm-tools v0.1.1
   Compiling uhyve-interface v0.1.3 (/home/runner/work/uhyve/uhyve/uhyve-interface)
   Compiling zerocopy v0.7.35
   Compiling rftrace-frontend v0.2.3
   Compiling rftrace v0.2.3
error[E0554]: `#![feature]` may not be used on the stable release channel
 --> /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rftrace-frontend-0.2.3/src/lib.rs:5:1
  |
5 | #![feature(vec_into_raw_parts)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0554`.
```

I attempted to implement this in https://github.com/hermit-os/uhyve/pull/991, but the CI fails against `stable` for the reason shown above. Therefore, this is blocked until https://github.com/rust-lang/rust/issues/65816 stabilizes.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -26,6 +26,22 @@ jobs:
     env:
       RUST_BACKTRACE: 1
 
+  test-features:
+    name: Test all features
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - uses: actions/checkout@v7
+      - uses: dtolnay/rust-toolchain@nightly
+        with:
+          targets: x86_64-unknown-none
+          components: llvm-tools
+      - name: Test
+        run: |
+          cargo test --lib --all-features
+    env:
+      RUST_BACKTRACE: 1
+
   integration-tests:
     name: Integration Tests
     runs-on: [self-hosted]

